### PR TITLE
[fix][cli] Fulfill add-opens to function-localrunner also

### PR DESCRIPTION
**Pull request by tisonkun**
This closes https://github.com/apache/pulsar/issues/20282.

### Motivation

Fulfill add-opens to function-localrunner also.

This issue affects version >= 2.11.

### Verifying this change

- [x] Make sure that the change passes the CI checks and manually tested locally.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: bin/function-localrunner [hunk 1]
```diff
@@ -40,13 +40,15 @@ PULSAR_MEM=${PULSAR_MEM:-"-Xmx128m -XX:MaxDirectMemorySize=128m"}
...
-if [[ -n "$IS_JAVA_8" ]]; then
-  PULSAR_GC_LOG=${PULSAR_GC_LOG:-"-Xloggc:logs/pulsar_gc_%p.log -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=10 -XX:GCLogFileSize=20M"}
-else
-# After jdk 9, gc log param should config like this. Ignoring version less than jdk 8
+IS_JAVA_8=$( $JAVA -version 2>&1 | grep version | grep '"1\.8' )
+if [[ -z "$IS_JAVA_8" ]]; then
+  # >= JDK 9
   PULSAR_GC_LOG=${PULSAR_GC_LOG:-"-Xlog:gc:logs/pulsar_gc_%p.log:time,uptime:filecount=10,filesize=20M"}
+  # '--add-opens' option is not supported in JDK 1.8
+  OPTS="$OPTS --add-opens java.base/sun.net=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED"
```
Comment (tisonkun) [inline bin/function-localrunner:48]: This is the effective changes. Others are refactors during the debugging stage.

**Comment by lifepuzzlefun** [comment]
@tisonkun thanks for this fix. and i saw more add-open options in pulsar start script. do we need them also ? see https://github.com/apache/pulsar/blob/aa3bfcda6935dc2d9c69dce51f96ba80baeae0b7/bin/pulsar#L297-L311

**Comment by tisonkun** [comment]
@lifepuzzlefun I tend to add them on demand instead of pull in too much options. This is also how the upstream JDK's trend to restrict access.

That said, if we open unnecessary modules, it's more likely we introduce more dependency without explicit review.

**Comment by lifepuzzlefun** [comment]
> @lifepuzzlefun I tend to add them on demand instead of pull in too much options. This is also how the upstream JDK's trend to restrict access.
> 
> That said, if we open unnecessary modules, it's more likely we introduce more dependency without explicit review.

yes, i agree with you.

**Review by lifepuzzlefun — APPROVED** [review]

File: pulsar-common/src/test/java/org/apache/pulsar/common/util/netty/DnsResolverTest.java [hunk 2]
```diff
@@ -18,24 +18,29 @@
...
-import org.testng.Assert;
 import org.testng.annotations.Test;
 
 public class DnsResolverTest {
 
     @Test
-    public void testMaxTtl() {
-        EventLoop eventLoop = Mockito.mock(EventLoop.class);
-        DnsNameResolverBuilder dnsNameResolverBuilder = new DnsNameResolverBuilder(eventLoop);
+    public void testMaxTtl() throws Exception {
```
Comment (eolivelli) [inline pulsar-common/src/test/java/org/apache/pulsar/common/util/netty/DnsResolverTest.java:34]: please remove this change that seems unrelated

**Review by eolivelli — DISMISSED** [review]
Overall I support the patch, but there is some change that seems unrelated

**Comment by tisonkun** [comment]
@eolivelli reverted

**Review by nicoloboschi — APPROVED** [review]
